benchmark/Linear.hs

```diff
@@ -349,6 +349,8 @@ main =
         , benchIOSink "filter-all-out"  (Ops.filterAllOut 1)
         , benchIOSink "filter-all-in"   (Ops.filterAllIn 1)
         , benchIOSink "take-all"        (Ops.takeAll 1)
+        , benchIOSink "takeByTime: 10 sec" (Ops.takeByTime 10.0 1)
```

Comment:
1) 10 sec is very long, we keep benchmark times such that they complete quickly, this will hold up all benchmarks for too long. Usually more than 500 ms is too long because each benchmark also runs many samples and many iterations per sample. If the benchmark has to be long then we can put it under DEVBUILD or some such conditional.

2) Is this a benchmark? It does not seem to be measuring performance, it sounds more like a test rather than benchmark.

Reply:
The benchmark should perhaps measure how many elements it can take in the given amount of time.